### PR TITLE
feat(chore): migrate to use npm workspaces

### LINKED ISSUE #1673: Migrating to Use npm Workspaces
Is your feature request related to a problem? Please describe.
1.In lerna v7.0.0,  Lerna is not responsible for installing and linking your dependencies in repo, your package manager is much better suited to that task.
2.Modify the script  **(Referencing to https://github.com/loopbackio/loopback-next/issues/9703)**  where we need to use a custom eslint configuration within your package to  decouple package's eslint usage from the LoopBack build eslint  and other improvements. 

Describe the solution you'd like

The way to achieve this is by using npm workspaces feature while keeping Lerna for publishing and version management
- npm (https://docs.npmjs.com/cli/using-npm/workspaces)

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
Migrate to use npm workspaces

BREAKING CHANGE:
lerna bootstrap command will not be needed as npm will  manage the same functionality.

GH-1673

## Description
Migrate to use npm workspaces

Fixes #1673 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sandbox/chat-notification-socketio-example/services/notifications-service/src/application.ts [hunk 1]
```diff
@@ -52,9 +52,6 @@ export class ChatAndNotifApplication extends BootMixin(
     this.component(NotificationServiceComponent);
 
     this.bind(NotifServiceBindings.Config).to({
-      useCustomEmailProvider: true,
-      useCustomSMSProvider: true,
-      useCustomPushProvider: true,
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply sandbox/chat-notification-socketio-example/services/notifications-service/src/application.ts:57]: updated version in sandbox notification example

File: package.json [hunk 2]
```diff
@@ -27,5 +27,14 @@
...
+  },
+  "workspaces": [
+    "packages/core/",
+    "packages/cli/",
+    "packages/cache",
+    "packages/feature-toggle",
+    "packages/custom-sf-changelog/",
+    "services/*",
+    "sandbox/**"
+  ]
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply package.json:39]: Done

**Review by yeshamavani — APPROVED** [review]

File: services/audit-service/src/services/job-processing.service.ts [hunk 3]
```diff
@@ -69,7 +69,9 @@ export class JobProcessingService {
           (customFilter.actedOn == null ||
             filterUsed.actedOn == null ||
             filterUsed.actedOn === customFilter.actedOn) &&
+          // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
```
Comment (samarpan-b) [inline services/audit-service/src/services/job-processing.service.ts:72]: remove this disable eslint and fix the issue

File: services/bpmn-service/package.json [hunk 4]
```diff
@@ -1,135 +1,135 @@
...
+    "version": "7.0.8",
+    "description": "BPMN microservice",
+    "keywords": [
+        "loopback-extension",
+        "loopback"
+    ],
+    "main": "dist/index.js",
+    "types": "dist/index.d.ts",
+    "engines": {
+        "node": "16 || 17 || 18"
```
Comment (samarpan-b) [inline services/bpmn-service/package.json:12]: `"engines": {
    "node": "16 || 18 || 20"
  },`

File: services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts [hunk 5]
```diff
@@ -41,10 +41,10 @@ describe('Group Controller', function (this: Mocha.Suite) {
     tenantId: id,
     password: pass,
     permissions: [
-      PermissionKey.ViewUserGroupList,
-      PermissionKey.UpdateUserGroup,
-      PermissionKey.DeleteUserGroup,
-      PermissionKey.CreateUserGroup,
+      PermissionKey.ViewGroupList,
```
Comment (samarpan-b) [inline services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts:44]: Why this change ?

File: services/user-tenant-service/src/enums/permission-key.enum.ts [hunk 6]
```diff
@@ -10,10 +10,10 @@ export const enum PermissionKey {
   ViewUserGroupList = 'ViewUserGroupList',
   UpdateUserGroup = 'UpdateUserGroup',
   DeleteUserGroup = 'DeleteUserGroup',
-  CreateGroup = 'CreateUserGroup',
-  ViewGroupList = 'ViewUserGroupList',
-  UpdateGroup = 'UpdateUserGroup',
-  DeleteGroup = 'DeleteUserGroup',
+  CreateGroup = 'CreateGroup',
```
Comment (samarpan-b) [inline services/user-tenant-service/src/enums/permission-key.enum.ts:13]: why this change ?

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts (hunk 5 above)
Comment (Surbhi-sharma1) [reply services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts:44]: Was getting lint issue of duplicate enum values

File: services/user-tenant-service/src/enums/permission-key.enum.ts (hunk 6 above)
Comment (samarpan-b) [reply services/user-tenant-service/src/enums/permission-key.enum.ts:13]: ok

File: services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts (hunk 5 above)
Comment (samarpan-b) [reply services/user-tenant-service/src/__tests__/acceptance/tenant-group.controller.acceptance.ts:44]: ok

**Review by shivamSF — CHANGES_REQUESTED** [review]

File: .prettierrc [hunk 7]
```diff
@@ -5,3 +5,10 @@
   "trailingComma": "all",
   "arrowParens": "avoid"
 }
+{
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply .prettierrc:8]: ok

File: package.json [hunk 8]
```diff
@@ -27,5 +27,13 @@
   "dependencies": {
     "handlebars": "^4.7.7",
     "patch-package": "^8.0.0"
-  }
+  },
+  "workspaces": [
+    "packages/core/",
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply package.json:32]: Ocr-parser is  excluded as its having angular dependency

**Review by shivamSF — APPROVED** [review]

File: packages/cli/tsconfig.json [hunk 9]
```diff
@@ -8,7 +9,11 @@
     "strict": true,
     "target": "es2017",
     "composite": false,
-    "typeRoots": ["./src/types", "./node_modules/@types"],
+    "typeRoots": [
+      "./../../node_modules/@types",
+      "./node_modules/@types",
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply packages/cli/tsconfig.json:14]: @types/jasmine and mocha was giving path related issues  in Cli build.

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**